Zend/zend_execute.c

```diff
@@ -1463,6 +1478,17 @@ static ZEND_COLD void zend_verify_missing_return_type(const zend_function *zf)
 	zend_verify_return_error(zf, NULL);
 }
 
+bool zend_never_inline zend_verify_class_constant_type(const zend_class_constant *c, zval *constant)
+{
+	if (zend_check_type((zend_type *) &c->type, constant, NULL, NULL, /* is_return_type */ 1, /* is_internal_arg */ 0)) {
```

Comment:
Nit: Prefer true/false over 1/0.jj

Reply:
`(zend_type *)` cast is unnecessary, right?

Reply:
Ah, I didn't notice the unnecessary cast until now. I'm not sure why it was needed (since I took over the original implementation). Nevertheless, I'll get rid of it.